Meta/CMake/serenity_options.cmake

```diff
@@ -10,3 +10,4 @@ serenity_option(ENABLE_KERNEL_ADDRESS_SANITIZER OFF CACHE BOOL "Enable kernel ad
 serenity_option(ENABLE_KERNEL_COVERAGE_COLLECTION  OFF CACHE BOOL "Enable KCOV and kernel coverage instrumentation in gcc/clang")
 serenity_option(ENABLE_KERNEL_LTO OFF CACHE BOOL "Build the kernel with link-time optimization")
 serenity_option(ENABLE_EXTRA_KERNEL_DEBUG_SYMBOLS  OFF CACHE BOOL "Enable -Og and -ggdb3 options for Kernel code for easier debugging")
+serenity_option(USE_MOLD_LINKER OFF CACHE BOOL "Link the SerenityOS userland with the mold linker")
```

Comment:
I checked and we don't have another option in the source tree with the pattern `USE_<SOMETHING>`. Maybe this should be called `ENABLE_MOLD_LINKER` or something similar instead? That would match our existing patterns better than the current option name.